src/main/java/uk/gov/ch/oauth/Oauth2.java

```diff
@@ -187,4 +193,13 @@ OAuthToken requestOAuthToken(String code) {
                 .bodyToMono(OAuthToken.class);
         return postReq.block(timeoutDuration);
     }
+
+    private void rengenrateSessionID(HttpServletResponse httpServletResponse) {
+        Session chSession = sessionFactory.getSessionFromContext();
+        LOGGER.debug("Original Session ID: " + chSession.getCookieId());
+        final Map<String, Object> originalSessionData = chSession.getData();
+        Session session = sessionFactory
+                .regenerateSession(originalSessionData, chSession);
```

Comment:
Since the `originalSessionData` is a property of the `chSession` that is passed to `regenerateSession`, wouldn't it be better to get it inside the method, and only pass in `chSession` to reduce coupling?